p2p-tests/contributor-fork-replication-2-test.ts

```diff
@@ -0,0 +1,203 @@
+// we add a commit to the project on contributor while not connected to the
+// seed.
+async function testcase(dataPath: string) {
+  const project = {
+    name: "my-fancy-project",
+    urn: "rad:git:hnrkrhhs1goaawo7db1gpyct8hd7mif5q8c3o",
+  };
+
+  const seed = new UpstreamSeed({
+    name: "seed",
+    ipAddress: "10.0.0.1",
+    project: project.urn,
+    dataPath,
+  });
+
+  const maintainer = new RadicleProxy({
+    name: "maintainer",
+    ipAddress: "10.0.0.101",
+    seed: seed.seedAddress,
+    dataPath,
+  });
+
+  const contributor = new RadicleProxy({
+    name: "contributor",
+    ipAddress: "10.0.0.102",
+    seed: seed.seedAddress,
+    dataPath,
+  });
+
+  maintainer.start();
+
+  // Maintainer creates a new project.
+  await withRetry(async () => {
+    await maintainer.proxyClient.project.create({
+      repo: {
+        type: "new",
+        path: maintainer.checkoutPath,
+        name: project.name,
+      },
+      description: "",
+      defaultBranch: "main",
+    });
+  });
+
+  await maintainer.stop();
+  seed.start();
+  maintainer.start();
+
+  // Assert that the seed received the project.
+  await withRetry(async () => {
+    const result = radCli({
+      radHome: seed.radHome,
+      args: ["identities", "project", "get", "--urn", project.urn],
+    });
+
+    strictAssert.deepStrictEqual(result, {
+      urn: project.urn,
+      payload: {
+        "https://radicle.xyz/link/identities/project/v1": {
+          name: "my-fancy-project",
+          description: "",
+          default_branch: "main",
+        },
+      },
+    });
+  });
+
+  // Without this the test fails, not sure why.
+  await sleep(1000);
+
+  await maintainer.stop();
+  contributor.start();
+
+  // Contributor follows the project.
+  await withRetry(async () => {
+    await contributor.proxyClient.project.requestSubmit(project.urn);
+  });
+
+  // Assert that the contributor received the project.
+  await withRetry(async () => {
+    const result = await contributor.proxyClient.project.get(project.urn);
+
+    strictAssert.deepStrictEqual(result, {
+      urn: project.urn,
+      metadata: {
+        name: project.name,
+        description: "",
+        defaultBranch: "main",
+        maintainers: [maintainer.identityUrn],
+      },
+      stats: { commits: 1, branches: 0, contributors: 1 },
+    });
+  });
+
+  // Contributor forks the project.
+  const contributorCheckoutPath =
+    await contributor.proxyClient.project.checkout(project.urn, {
+      path: contributor.checkoutPath,
+      peerId: maintainer.peerId,
+    });
+
+  await seed.stop();
+  // Contributor publishes a new commit.
+  commit({
+    author: contributor.name,
+    checkoutPath: contributorCheckoutPath,
+  });
+  pushRad({
+    radHome: contributor.radHome,
+    checkoutPath: contributorCheckoutPath,
+    keyPassphrase: contributor.passphrase,
+  });
+  await sleep(1000);
+  await contributor.stop();
+
+  seed.start();
+  await sleep(1000);
```

Comment:
Is this necessary?

Reply:
bb4bb06aefe64791ff66e07e4d400eba855a3a9e